File: 2126/main.py
caso = 1
while True:
    try:
        n1 = input()
        n2 = input()
        print('caso #%d:' % caso)
        menge = n2.count(n1)
        if menge == 0:
            print('Nao existe subsequencia')
        else:
            print('menged.Subsequencias: %d' % menge)
            menge = n2.rfind(n1)
            print('Pos: %d' % (int(menge)+1))
        print()
        caso += 1
        
    except EOFError:
        break

File: 2126/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	caseNumber := 1

	for {
		fmt.Printf("Caso #%d:\n", caseNumber)

		// Ler o primeiro número
		if !scanner.Scan() {
			break
		}
		n1 := scanner.Text()

		// Ler o segundo número
		if !scanner.Scan() {
			break
		}
		n2 := scanner.Text()

		// Contar as subsequências
		qtd := strings.Count(n2, n1)
		if qtd == 0 {
			fmt.Println("Nao existe subsequencia")
		} else {
			fmt.Printf("Qtd.Subsequencias: %d\n", qtd)
			pos := strings.LastIndex(n2, n1) + 1
			fmt.Printf("Pos: %d\n", pos)
		}

		fmt.Println()
		caseNumber++
	}
}
